source/adaptive_load/nighthawk_service_client_impl.cc

```diff
@@ -0,0 +1,46 @@
+#include "adaptive_load/nighthawk_service_client_impl.h"
+
+#include "external/envoy/source/common/common/assert.h"
+
+namespace Nighthawk {
+
+absl::StatusOr<nighthawk::client::ExecutionResponse>
+NighthawkServiceClientImpl::PerformNighthawkBenchmark(
+    nighthawk::client::NighthawkService::StubInterface* nighthawk_service_stub,
+    const nighthawk::client::CommandLineOptions& command_line_options,
+    const Envoy::Protobuf::Duration& duration) {
```

Comment:
This seems like an odd api to me. Accepting a const command_line_options reference and then immediately copying it into a new object feels counterproductive.

There also doesn't seem to me to be a good reason to have duration as its own parameter as opposed to set on the command_line_options here, since the first and only thing you do with duration is set it in options.  Could you accept a const reference to command line options, with the duration properly included, and then use that in ExecutionRequest?

Reply:
This API has always bothered me a bit. It made more sense when it was a private helper and I was trying to save lines at the call sites. Changing this as you suggest.